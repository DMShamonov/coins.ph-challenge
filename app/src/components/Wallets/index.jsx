// @flow

import * as React from 'react';

import {
  Table,
  TableHeader,
  TableBody,
  TableRow,
  TableCell,
} from 'components/Table';
import Wallet from 'components/Wallet';

import type { PropsType } from './types.js.flow';

export default function Wallets({
  wallets,
}: PropsType): React.Node {
  return (
    <Table>
      <TableHeader>
        <TableRow>
          <TableCell>ID</TableCell>
          <TableCell>Label</TableCell>
          <TableCell width="10%">Balance</TableCell>
        </TableRow>
      </TableHeader>
      <TableBody>
        {wallets.map(({ id, label, balance }) => (
          <Wallet id={id} label={label} balance={balance} key={id} />
        ))}
      </TableBody>
    </Table>
  );
}
