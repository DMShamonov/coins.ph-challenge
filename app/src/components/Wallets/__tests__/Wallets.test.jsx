import React from 'react';
import { shallow } from 'enzyme';

import Wallets from '../index';

describe('`Wallets` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Wallets wallets={[{ id: '__ID__', label: '__LABEL__', balance: 100000000 }]} />)).toMatchSnapshot();
  });

  test('when `wallets` prop is empty array', () => {
    expect(shallow(<Wallets wallets={[]} />)).toMatchSnapshot();
  });
});
