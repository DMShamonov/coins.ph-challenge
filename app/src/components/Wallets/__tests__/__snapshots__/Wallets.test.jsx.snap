// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Wallets\` component should render correctly when \`wallets\` prop is empty array 1`] = `
<Table>
  <TableHeader>
    <TableRow>
      <TableCell>
        ID
      </TableCell>
      <TableCell>
        Label
      </TableCell>
      <TableCell
        width="10%"
      >
        Balance
      </TableCell>
    </TableRow>
  </TableHeader>
  <TableBody />
</Table>
`;

exports[`\`Wallets\` component should render correctly with default props 1`] = `
<Table>
  <TableHeader>
    <TableRow>
      <TableCell>
        ID
      </TableCell>
      <TableCell>
        Label
      </TableCell>
      <TableCell
        width="10%"
      >
        Balance
      </TableCell>
    </TableRow>
  </TableHeader>
  <TableBody>
    <Wallet
      balance={100000000}
      id="__ID__"
      key="__ID__"
      label="__LABEL__"
    />
  </TableBody>
</Table>
`;
